### PR TITLE
Connector service refactor

### PR DESCRIPTION
**Pull request by wangch079**
The changes made by this PR:
1. Move `connectors_service` and `list_connectors` to `lib/app`, as suggested by @artem-shelkovnikov here: https://github.com/elastic/connectors-ruby/pull/318#issuecomment-1251009785
2. Update `make run`
3. Rename `start_polling_jobs!` to `start_polling_tasks!`
4. Rename `non-native mode` to `single mode`

#### Pre-Review Checklist
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] (Elastic Only) Built gems (both `connectors_utility` and `connectors_service`) and included into Enterprise Search and tested that Enterprise Search works well with new gem versions.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -78,7 +78,7 @@ run-docker:
 	docker run --env "elasticsearch.hosts=http://host.docker.internal:9200" --env "elasticsearch.api_key=$(API_KEY)" --rm -it connectors
 
 exec_app:
-	cd lib/app; bundle _$(shell cat .bundler-version)_ exec ruby app.rb
+	cd lib/app; bundle _$(shell cat .bundler-version)_ exec ruby -r "./application.rb" -e "App::Application.run!"
```
[the remark this answers is not in the document]
Comment (tarekziade) [reply Makefile:81]: can't you use `bin/connectors_service` here?

[the remark this answers is not in the document]
Comment (wangch079) [reply Makefile:81]: Good idea, will try

**Comment by tarekziade** [comment]
I get this 

I remember I had to add `rack` when installing the coservice, https://github.com/elastic/ent-search/blob/main/dist/shared/cloud/coservices/install-services.sh.erb#L46

```
Dev/connectors-ruby % make run
"yq" e ".revision = \"199aac5503abd7430e1216d34107328553a2ae32\"" -i config/connectors.yml
"yq" e ".repository = \"https://github.com/elastic/connectors-ruby.git\"" -i config/connectors.yml
"yq" e ".version = \"8.6.0.1-20220927T034701Z\"" -i config/connectors.yml
cd lib/app; bundle _2.3.15_ exec ruby -r "./application.rb" -e "App::Application.run!"
Parsing /Users/tarekziade/Dev/connectors-ruby/lib/app/../../config/connectors.yml configuration file.
{"@timestamp":"2022-09-27T03:47:02.978Z","log.level":"INFO","message":"ENT_SEARCH_CONFIG_PATH is not found, use connector service config.","ecs.version":"1.4.0","labels":{"index_date":"2022.09.26"},"log":{"logger":null},"service":{"type":"connectors-ruby","version":"8.6.0.1-20220927T034701Z"},"process":{"pid":1435,"name":"-e","thread":70255082938400}}
/Users/tarekziade/.rbenv/versions/2.6.9/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- rack/utils (LoadError)
	from /Users/tarekziade/.rbenv/versions/2.6.9/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /Users/tarekziade/Dev/connectors-ruby/lib/connectors/gitlab/extractor.rb:11:in `<top (required)>'
	from /Users/tarekziade/.rbenv/versions/2.6.9/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /Users/tarekziade/.rbenv/versions/2.6.9/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /Users/tarekziade/Dev/connectors-ruby/lib/connectors/gitlab/connector.rb:11:in `<top (required)>'
	from /Users/tarekziade/.rbenv/versions/2.6.9/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /Users/tarekziade/.rbenv/versions/2.6.9/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /Users/tarekziade/Dev/connectors-ruby/lib/connectors.rb:16:in `block in <top (required)>'
	from /Users/tarekziade/Dev/connectors-ruby/lib/connectors.rb:16:in `each'
	from /Users/tarekziade/Dev/connectors-ruby/lib/connectors.rb:16:in `<top (required)>'
	from /Users/tarekziade/.rbenv/versions/2.6.9/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /Users/tarekziade/.rbenv/versions/2.6.9/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /Users/tarekziade/Dev/connectors-ruby/lib/app/dispatcher.rb:11:in `<top (required)>'
	from /Users/tarekziade/.rbenv/versions/2.6.9/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /Users/tarekziade/.rbenv/versions/2.6.9/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /Users/tarekziade/Dev/connectors-ruby/lib/app/application.rb:11:in `<top (required)>'
	from /Users/tarekziade/.rbenv/versions/2.6.9/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /Users/tarekziade/.rbenv/versions/2.6.9/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
make: *** [exec_app] Error 1
```

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (artem-shelkovnikov) [reply Makefile:81]: @wangch079 are you planning on finishing this PR?

[the remark this answers is not in the document]
Comment (wangch079) [reply Makefile:81]: Yes, But I have more urgent work to tackle. As this won't go into 8.5, I leave it to the bottom of my TODO list

**Review by artem-shelkovnikov — APPROVED** [review]

**Comment by mchernyavskaya** [comment]
@wangch079 is this still relevant? I see it's been approved but still not merged.

**Comment by wangch079** [comment]
Close for now. Only refactor of Makefile is worth looking at, and will be done later.

**Pull request closed without merging**